### PR TITLE
Add unzip shell script

### PR DESCRIPTION
**Pull request by akinny**
Move unzip to script to work for both windows and linux
shorten property names by static import
Fix shell scripts

Before you contribute, please review these guidelines to help ensure a smooth process for everyone.

Thanks.

* Read [how to properly contribute to open source projects on Github][1].
* Fork the project.
* Use a feature branch.
* Write [good commit messages][2].
* Use the same coding conventions as the rest of the project.
* Commit locally and push to your fork until you are happy with your contribution.
* Make sure to add tests and verify all the tests are passing when merging upstream.
* Add an entry to the [Changelog][3] accordingly.
* Please add your name to the [CONTRIBUTORS.md][7] file. Adding your name to the [CONTRIBUTORS.md][7] file signifies agreement to all rights and reservations provided by the [License][4].
* [Squash related commits together][5].
* Open a [pull request][6].
* The pull request will be reviewed by the community and merged by the project committers.

[1]: http://gun.io/blog/how-to-github-fork-branch-and-pull-request
[2]: http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[3]: ./CHANGELOG.md
[4]: ./LICENSE
[5]: http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html
[6]: https://help.github.com/articles/using-pull-requests
[7]: ./CONTRIBUTORS.md

### CODE REVIEW

**Review by PeterTeunissen — COMMENTED** [review]
Looks good to me

File: jwala-services/src/main/java/com/cerner/jwala/service/binarydistribution/impl/BinaryDistributionServiceImpl.java [hunk 1]
```diff
@@ -199,21 +201,31 @@ public boolean remoteFileCheck(final String hostname, final String remoteFilePat
...
         LOGGER.info("Start deploy unzip for {}", hostname);
         final String jwalaScriptsPath = ApplicationProperties.getRequired(PropertyKeys.REMOTE_SCRIPT_DIR);
         if (!remoteFileCheck(hostname, jwalaScriptsPath)) {
             remoteCreateDirectory(hostname, jwalaScriptsPath);
         }
-
         if (!remoteFileCheck(hostname, jwalaScriptsPath + "/" + UNZIPEXE)) {
             final String unzipFile = ApplicationProperties.get(PropertyKeys.LOCAL_JWALA_BINARY_DIR) + "/" + UNZIPEXE;
             LOGGER.info("SCP {} " + unzipFile);
             remoteSecureCopyFile(hostname, unzipFile,  jwalaScriptsPath + "/" + UNZIPEXE);
```
Comment (jmahmoodcc) [inline jwala-services/src/main/java/com/cerner/jwala/service/binarydistribution/impl/BinaryDistributionServiceImpl.java]: I know you didn't change this code but it's still relevant: does this work in linux? In the jwala-tomcat-linux.zip there is no unzip.exe in data/binaries so I would expect this to fail.

File: jwala-services/src/main/java/com/cerner/jwala/service/jvm/impl/JvmServiceImpl.java [hunk 2]
```diff
@@ -1,916 +1,918 @@
-package com.cerner.jwala.service.jvm.impl;
```
Comment (jmahmoodcc) [inline jwala-services/src/main/java/com/cerner/jwala/service/jvm/impl/JvmServiceImpl.java:1]: Why is the diff for this file showing as completely replaced? Did you make a lot of changes to JvmServiceImpl?

**Review by jmahmoodcc — COMMENTED** [review]
The unzip.sh part looks good to me - I would still look into the 2 comments I added

File: jwala-services/src/main/java/com/cerner/jwala/service/jvm/impl/JvmServiceImpl.java (hunk 2 above)
Comment (akinny) [reply jwala-services/src/main/java/com/cerner/jwala/service/jvm/impl/JvmServiceImpl.java:1]: No, few changes I think the line endings got changed to CRLF. You can see the diffenrence in intellij

File: jwala-services/src/main/java/com/cerner/jwala/service/binarydistribution/impl/BinaryDistributionServiceImpl.java (hunk 1 above)
Comment (akinny) [reply jwala-services/src/main/java/com/cerner/jwala/service/binarydistribution/impl/BinaryDistributionServiceImpl.java]: No the our windows unzip.exe will not work on linux. Linux comes with unzip sh the script will work. Good catch, I didn't realize we took it out of the bundle(which is good), I will add a check if file exists then scp, so I will not have to check host OS

File: jwala-services/src/main/java/com/cerner/jwala/service/binarydistribution/impl/BinaryDistributionControlServiceImpl.java [hunk 3]
```diff
@@ -136,14 +139,8 @@ private RemoteSystemConnection getConnection(String host) {
...
-    private String getUnzipCommand(String zipFileName, String destination){
-            if(zipFileName.indexOf(".zip")>-1){
-            return ApplicationProperties.getRequired(PropertyKeys.REMOTE_SCRIPT_DIR)+"/unzip.exe -q -o \"" + zipFileName + "\" -d \"" + destination + "\""; /*-x \" + aParams[3]"; TODO: exclude list*/
-        }else if(zipFileName.indexOf(".gz")>-1){
-            return String.format("tar xvf %s -C %s", zipFileName, destination);
-        }else{
-            throw new ApplicationException("Unknown zip file format");
-        }
+    private String getUnzipCommand(String zipFileName, String destination, String excludeFiles){
+            return ApplicationProperties.getRequired(PropertyKeys.REMOTE_SCRIPT_DIR)+"/"+ UNZIP_SCRIPT_NAME +" " + zipFileName + " " + destination + " " + excludeFiles;
```
Comment (jeddcuison) [inline jwala-services/src/main/java/com/cerner/jwala/service/binarydistribution/impl/BinaryDistributionControlServiceImpl.java]: Just a minor thing but the line looks better if the spaces are uniform

File: jwala-services/src/main/java/com/cerner/jwala/service/binarydistribution/impl/BinaryDistributionServiceImpl.java [hunk 4]
```diff
@@ -199,21 +201,31 @@ public boolean remoteFileCheck(final String hostname, final String remoteFilePat
...
     public void distributeUnzip(String hostname) {
         LOGGER.info("Start deploy unzip for {}", hostname);
         final String jwalaScriptsPath = ApplicationProperties.getRequired(PropertyKeys.REMOTE_SCRIPT_DIR);
         if (!remoteFileCheck(hostname, jwalaScriptsPath)) {
             remoteCreateDirectory(hostname, jwalaScriptsPath);
         }
-
         if (!remoteFileCheck(hostname, jwalaScriptsPath + "/" + UNZIPEXE)) {
             final String unzipFile = ApplicationProperties.get(PropertyKeys.LOCAL_JWALA_BINARY_DIR) + "/" + UNZIPEXE;
             LOGGER.info("SCP {} " + unzipFile);
```
Comment (jeddcuison) [inline jwala-services/src/main/java/com/cerner/jwala/service/binarydistribution/impl/BinaryDistributionServiceImpl.java]: Please just change '+' to  ','

**Review by jeddcuison — COMMENTED** [review]
Looks good, I just have 2 comments and the other one is minor

**Review by jeddcuison — APPROVED** [review]

**Review by jmahmoodcc — APPROVED** [review]

**Review by jmahmoodcc — APPROVED** [review]

**Pull request merged**